test/pytest/test_bridge.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import asyncio
+import json
+import sys
+import unittest
+
+from typing import Any, Dict, Tuple
+
+import systemd_ctypes
+from cockpit.router import Router
+
+asyncio.set_event_loop_policy(systemd_ctypes.EventLoopPolicy())
+
+
+class MockTransport(asyncio.Transport):
+    queue: asyncio.Queue[Tuple[str, bytes]]
+
+    def send_json(self, _channel: str, **kwargs) -> None:
+        self.send_data(_channel, json.dumps(kwargs).encode('ascii'))
+
+    def send_data(self, channel: str, data: bytes) -> None:
+        msg = channel.encode('ascii') + b'\n' + data
+        msg = str(len(msg)).encode('ascii') + b'\n' + msg
+        self.protocol.data_received(msg)
+
+    def send_init(self, version=1, host='test', **kwargs):
+        self.send_json('', command='init', version=version, host=host, **kwargs)
+
+    def send_open(self, channel, payload, **kwargs):
+        self.send_json('', command='open', channel=channel, payload=payload, **kwargs)
```

Comment:
This looks weird. New channels are always opened on the control channel AFAIK. Why does this need a channel name? Moreover, why does it specify channel twice? Once in the first positional arg (which looks like the correct one), and a different one as kwarg?

Reply:
See the comment above.  The "channel" in the sense of the framing protocol has to be the empty string for control messages.  Separately, the control message, itself, has a (non-empty) "channel" field.